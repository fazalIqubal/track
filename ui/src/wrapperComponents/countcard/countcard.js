import React from 'react';
import './countcard.scss';
import {Row, Col, Card, Statistic} from 'antd'

export class CountCard extends React.Component{
    render(){
        return(
            <Card>
                <Row>
                    <Col span={12}>
                        <Statistic
                            className={this.props.className}
                            value={this.props.value}
                            title={this.props.title}
                        />
                    </Col>
                    <Col span={12} className='card-icon'>
                       <img src= {this.props.icon} className="icon-img" />
                    </Col>
                </Row>
            </Card> 
            
        )
    }
}
