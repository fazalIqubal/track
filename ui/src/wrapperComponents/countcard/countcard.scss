.card-icon{
    text-align: right;
    margin: auto;

    .icon-img{
        width: 35%;
    }
}
