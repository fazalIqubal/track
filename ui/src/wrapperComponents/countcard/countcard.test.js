import React from 'react';
import {CountCard} from './countcard'; 
import { configure, shallow } from 'enzyme';
import { shallowToJson} from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

describe('should render correctly', () => {
    let cardOutput;
    beforeEach(() => {
        cardOutput = shallow(<CountCard/>
        );
    });
    afterEach(() => {
        cardOutput = null;
    });

    it('should render correctly ', () => {
        expect(shallowToJson(cardOutput)).toMatchSnapshot()
    });

    it('contains an <Card/> component', () => {
        expect(cardOutput.find('Card').exists()).toEqual(true);
    });

    it('should have icon', () => {
        expect(cardOutput.find('img').exists()).toEqual(true);
    });

})