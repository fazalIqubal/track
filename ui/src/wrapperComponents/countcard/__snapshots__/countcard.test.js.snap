// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly should render correctly  1`] = `
<Card>
  <Row>
    <Col
      span={12}
    >
      <withConfigConsumer(Statistic) />
    </Col>
    <Col
      className="card-icon"
      span={12}
    >
      <img
        className="icon-img"
      />
    </Col>
  </Row>
</Card>
`;
